src/jsMain/kotlin/io/beatmaps/index/beatmapInfo.kt

```diff
@@ -94,8 +93,12 @@ class BeatmapInfo : RComponent<BeatmapInfoProps, BeatMapInfoState>() {
 
                     div {
                         val totalVotes = (map.stats.upvotes + map.stats.downvotes).toDouble()
-                        val rawScore = map.stats.upvotes / totalVotes
-                        val uncertainty = abs((rawScore - 0.5) * 2.0.pow(-log10(totalVotes + 1)))
+                        var uncertainty = 2.0.pow(-log(totalVotes / 2 + 1, 3.0))
+                        val weightedRange = 25
+                        val weighting = 2
+                        if ((totalVotes + weighting) < weightedRange) {
+                            uncertainty += (1 - uncertainty) * (1 - (totalVotes + weighting) * (1 / weightedRange))
```

Comment:
`(1 / weightedRange)` will use integer division and always be 0, making uncertainty always 1. You then get a lovely division by 0 below. All of which means any map under 23 votes has no grey bar at all

Reply:
A classic textbook mistake. 🤦